Sprint-2/extend/format-time.js

```diff
@@ -3,10 +3,16 @@
 
 function formatAs12HourClock(time) {
   const hours = Number(time.slice(0, 2));
-  if (hours > 12) {
-    return `${hours - 12}:00 pm`;
+
+  if (hours === 0) {
+    return `12:00 am`; // Midnight
+  } else if (hours === 12) {
+    return `12:00 pm`; // Noon 
+  } else if (hours > 12) {
+    return `${hours - 12}:00 pm`; // Afternoon
+  } else {
+    return `0${hours}:00 am`; // Morning 
   }
```

Comment:
Well done, this works well in the test cases described. Can you think about how we can improve this to handle inputs such as `formatAs12HourClock("7:30")` for example? 
Consider what other ways you might be able to split the string, where you can be certain which part is the hours and which part is the minutes.
Add test to assert that these inputs will be handled correctly.

Reply:
Thank you for the feedback, Yoan! I propose splitting the input string into hours and minutes based on the colon (":") to address your point. I also added some tests to assert that these inputs will be handled correctly.